hyper_click/generic_path_resolver.py

```diff
@@ -150,7 +150,7 @@ def resolve_as_file(self, path_name):
         if path.isfile(path_name):
             return path_name
 
-        with_ext = self.resolve_with_exts(path_name)
+        with_ext = self.resolve_with_exts(path_name) or self.resolve_with_exts(path.join(path_name, 'index'))
```

Comment:
@iahu is this necessary? There is already `index` file check in `resolve_index` method.